### PR TITLE
Test covergage

### LINKED ISSUE #21: better test coverage
increase test coverage and set up coverage report on CI

### PR DESCRIPTION
**Pull request by Olthoff231381**
Included code coverage and better tests with exceptions
closes #21

### CODE REVIEW

**Comment by Olthoff231381** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=ssciwr_mailcom&pullRequest=29) **Quality Gate failed**
> 
> Failed conditions  [D Reliability Rating on New Code](https://sonarcloud.io/dashboard?id=ssciwr_mailcom&pullRequest=29) (required ≥ A)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=ssciwr_mailcom&pullRequest=29)
> 
>  Catch issues before they fail your Quality Gate with our IDE extension  [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)

Missed the assertion in the test. Is there a chance to include it in this PR without much problem?

File: mailcom/inout.py [hunk 1]
```diff
@@ -8,6 +8,8 @@ def list_of_files(directory_name: str) -> list[Path]:
     mypath = Path(directory_name)
     pattern = [".eml", ".html"]  # we would not change the file type through user input
     email_list = [mp.resolve() for mp in mypath.glob("**/*") if mp.suffix in pattern]
+    if len(email_list) == 0:
+        raise ValueError("The directory {} does not contain .eml or .html files".format(mypath))
```
Comment (iulusoy) [inline mailcom/inout.py:12]: Maybe add something like "Please check your input of the directory name containing the email data files" or the like, to be very explicit

File: mailcom/parse.py [hunk 2]
```diff
@@ -91,9 +91,11 @@ def init_transformers():
     return ner_recognizer
 
 
-def check_dir(path: str):
-    # check if directory is there
-    return os.path.exists(path) 
+def check_dir(path: str) -> bool:
+    if not os.path.exists(path):
+        raise OSError("Path {} does not exist".format(path))
```
Comment (iulusoy) [inline mailcom/parse.py:96]: This Error I would also add to the function `list_of_files`

File: mailcom/test/test_inout.py [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
+    assert len(list_of_files(tmp_path)) != 0
 
 def test_list_of_files_empty(tmp_path):
-    assert len(list_of_files(tmp_path)) == 0, "The list is not empty"
\ No newline at end of file
+    with pytest.raises(ValueError):
+        list_of_files(tmp_path)
+
+
+def test_list_of_files_correct_format(tmp_path):
```
Comment (iulusoy) [inline mailcom/test/test_inout.py:14]: nice test!

File: mailcom/test/test_inout.py [hunk 4]
```diff
@@ -1,9 +1,21 @@
 from mailcom.inout import list_of_files
+import pytest
 
 def test_list_of_files_found(tmp_path):
     p = tmp_path / "test.eml"
     p.write_text("test")
-    assert len(list_of_files(tmp_path)) != 0, "The list is empty"
+    assert len(list_of_files(tmp_path)) != 0
 
 def test_list_of_files_empty(tmp_path):
```
Comment (iulusoy) [inline mailcom/test/test_inout.py:9]: would then add the test for the exception if the provided input directory does not exist

**Review by iulusoy — CHANGES_REQUESTED** [review]
a few more items, looks good already.

**Review by iulusoy — APPROVED** [review]

**Pull request merged**